app/models/one_login/ruby_saml/saml_generator.rb

```diff
@@ -128,9 +128,9 @@ def set_attribute_values(attr_name, hbx_enrollment)
         when 'Assigned QHP Identifier'
           hbx_enrollment.product.hios_id.gsub('-', '')
         when 'Total Amount Owed'
-          hbx_enrollment.total_premium - hbx_enrollment.applied_aptc_amount.to_f
+          (hbx_enrollment.total_premium.to_f - hbx_enrollment.applied_aptc_amount.to_f).round(2)
         when 'Premium Amount Total'
-          hbx_enrollment.total_premium
+          hbx_enrollment.total_premium.to_f.round(2)
```

Comment:
Also, this is not needed as the total_premium returns a float value rounded to 2 decimals